src/Microsoft.VisualStudio.Composition/CompositionConfiguration.cs

```diff
@@ -150,19 +142,49 @@ public static CompositionConfiguration Create(ComposableCatalog catalog)
             // Detect loops of all non-shared parts.
             errors.AddRange(FindLoops(parts));
 
+            // If errors are found, re-validate the salvaged parts in case there are parts whose dependencies are affected by the initial errors
             if (errors.Count > 0)
             {
-                var invalidParts = ImmutableHashSet.CreateRange(errors.SelectMany(error => error.Parts).Select(p => p.Definition));
-                if (invalidParts.IsEmpty)
+                // Set salvaged parts to current catalog in case there are no errors
+                var salvagedParts = parts;
+                var salvagedPartDefinitions = catalog.Parts;
+
+                List<ComposedPartDiagnostic> previousErrors = errors;
+                Stack<IReadOnlyCollection<ComposedPartDiagnostic>> stackedErrors = new Stack<IReadOnlyCollection<ComposedPartDiagnostic>>();
+
+                // While we still find errors we validate the exports so that we remove all dependency failures
+                while (previousErrors.Count > 0)
                 {
-                    // If we can't identify the faulty parts but we still have errors, we have to just throw.
-                    throw new CompositionFailedException(Strings.FailStableComposition, ImmutableStack.Create<IReadOnlyCollection<ComposedPartDiagnostic>>(errors));
```

Comment:
I don't see an equivalent place where you moved this code. Shouldn't we still notice and throw when a composition is invalid without identified parts?